File: practice-2025-10-18/g-fading-wind/hugo.cpp
// Link: https://open.kattis.com/problems/fadingwind
//
// Constraints:
//
//    Doesn't matter but every input is (1 <= n <= 10^3)
//

#include <iostream>
#include <cmath>

int main() {
    int v = 0, h = 0, k = 0, s = 0, d = 0;
    
    std::cin >> h >> k >> v >> s;
    
    while (h > 0) {
        v += s;
        
        v -= std::max(1, int(std::floor(v / 10)));
        
        if (v >= k) {
            h += 1;
        }
        
        if (0 < v && v < k) {
            h -= 1;
            if (h == 0) {
                v = 0;
            }
        }
        
        if (v <= 0) {
            h = 0;
            v = 0;
        }
        
        d += v;
        if (s > 0) {
            s -= 1;
        }
    }
    
    std::cout << d;
    
    return 0;
}

File: practice-2025-10-18/c-champernowne-count/hugo.py
'''
    Link: https://open.kattis.com/problems/champernownecount

    Constraints:

        (1 <= n <= 10^5)
        (1 <= k <= 10^9)
'''

n, k = map(int, input().split())

count = 0
word_mod = 0

for i in range(1, n + 1):
    digits = len(str(i))

    word_mod = (word_mod * pow(10, digits, k) + i) % k

    if word_mod == 0:
        count += 1

print(count)